polkadot/node/subsystem-bench/src/lib/usage.rs

```diff
@@ -75,6 +76,44 @@ impl BenchmarkUsage {
 			_ => None,
 		}
 	}
+
+	pub fn save_as_json(&self, path: &str) -> std::io::Result<()> {
+		let mut file = File::create(path)?;
+		file.write_all(self.to_json().as_bytes())?;
+		Ok(())
+	}
+
+	fn to_json(&self) -> String {
+		let mut res: Vec<String> = vec![];
+		res.extend(
+			self.network_usage
+				.iter()
+				.map(|v| {
+					json!({
+						"name": v.resource_name,
+						"unit": "KiB",
+						"value": v.per_block
+					})
+					.to_string()
+				})
+				.collect::<Vec<_>>(),
+		);
+		res.extend(
+			self.cpu_usage
+				.iter()
+				.map(|v| {
+					json!({
+						"name": v.resource_name,
+						"unit": "seconds",
+						"value": v.per_block
+					})
+					.to_string()
+				})
+				.collect::<Vec<_>>(),
+		);
+
+		format!("[{}]", res.join(","))
```

Comment:
I think it would be better to use `serde_json` directly instead of custom creating JSON here.